File: mock-facility-management/src/main/java/de/vkb/dojo/es/facilityManagement/model/event/RoomLocked.java
package de.vkb.dojo.es.facilityManagement.model.event;

import com.fasterxml.jackson.annotation.*;

@JsonIgnoreProperties(ignoreUnknown = true)
@JsonTypeName("room-locked")
public class RoomLocked extends RoomEvent {
    private final String operationId;
    private final String aggregateId;

    @JsonCreator
    public RoomLocked(
            @JsonProperty("operatonId") String operationId,
            @JsonProperty("aggregateId") String aggregateId
    ) {
        this.operationId = operationId;
        this.aggregateId = aggregateId;
    }

    @Override
    @JsonGetter
    public String getOperationId() {
        return operationId;
    }

    @Override
    @JsonGetter
    public String getAggregateId() {
        return aggregateId;
    }
}

File: todo-reservation-micronaut/src/main/kotlin/de/vkb/dojo/es/facilityManagement/model/event/RoomLocked.kt
package de.vkb.dojo.es.facilityManagement.model.event

import com.fasterxml.jackson.annotation.JsonIgnoreProperties
import com.fasterxml.jackson.annotation.JsonTypeName

@JsonIgnoreProperties(ignoreUnknown = true)
@JsonTypeName("room-locked")
data class RoomLocked(
    override val operationId: String,
    override val aggregateId: String
): RoomEvent
